velox/dwio/common/ScanSpec.cpp

```diff
@@ -19,6 +19,20 @@
 
 namespace facebook::velox::common {
 
+ScanSpec* ScanSpec::getOrCreateChild(const std::string& name) {
+  auto container = this;
```

Comment:
No need to create this variable here.  You can also reuse this method in `getOrCreateChild(const Subfield&)`

Reply:
Updated to reuse this method. I find the variable `container` is needed as the return value. Am I missing something? Thanks.

Reply:
Thanks for your review. I got your point. Removed this variable and reused this method.

Reply:
The method `getOrCreateChild(const Subfield&)` creates child for a varying `container`, so `getOrCreateChild(const std::string& name)`, which creates child always for `this`, cannot be reused by it.

Reply:
You can call `container-> getOrCreateChild(nestedField->name())` in `getOrCreateChild(const Subfield&)`.  Then we can probably remove `ScanSpec(const Subfield::PathElement& element)` as well.

Reply:
Thanks for the pointer. Updated.